Lib/ensurepip/__init__.py

```diff
@@ -27,52 +17,36 @@
 _WHEEL_PKG_DIR = sysconfig.get_config_var('WHEEL_PKG_DIR')
 
 
-def _find_packages(path):
-    packages = {}
+def _find_wheel_pkg_dir_pip():
+    if _WHEEL_PKG_DIR is None:
+        return None
     try:
-        filenames = os.listdir(path)
+        filenames = os.listdir(_WHEEL_PKG_DIR)
     except OSError:
         # Ignore: path doesn't exist or permission error
-        filenames = ()
+        return None
     # Make the code deterministic if a directory contains multiple wheel files
     # of the same package, but don't attempt to implement correct version
     # comparison since this case should not happen.
-    filenames = sorted(filenames)
+    filenames = sorted(filenames, reverse=True)
     for filename in filenames:
```

Comment:
@AA-Turner if we do want to include this much refactoring in the PR, why no just use my pathlib approach that reads better without looping and going into the lower-level abstractions, being shorter and easier to grasp:
```python
dist_matching_wheels = Path(_WHEEL_PKG_DIR).glob('pip-*.whl')
try:
    last_matching_dist_wheel = sorted(dist_matching_wheels)[-1]
except IndexError as index_err:
    raise LookupError from index_err
filename = last_matching_dist_wheel.name
version = filename.removeprefix('pip-').partition('-')[0]
```
_(https://github.com/python/cpython/pull/12791/files#diff-f5e321563d5bc98371a960662f654a9387fd24f9bacb382d5b90d417ad2569c7R197-R205)_

Reply:
I like the ``glob`` approach (we could use ``glob.iglob`` or ``pathlib``).

Reply:
This is implemented in https://github.com/python/cpython/commit/b2a69a121a82d5abc8b99eed2492c4602d9fb91b, so I'm marking the thread as resolved.